views/components/Table/index.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+import {
+  Table as MuiTable,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  Box,
+  TableSortLabel,
+} from '@mui/material'
+import { visuallyHidden } from '@mui/utils'
+import { SORT_DIRECTION, fontSize } from '../../../constants'
+
+function EnhancedTableHead(props) {
+  const { sortDirection, sortProperty, onRequestSort, headCells } = props
+  const createSortHandler = (property) => (event) => {
+    onRequestSort(event, property)
+  }
+
+  return (
+    <TableHead>
+      <TableRow>
+        {headCells.map((headCell) => (
+          <TableCell
+            key={headCell.dataProperty}
+            sortDirection={
+              sortProperty === headCell.dataProperty ? sortDirection : false
+            }
+            sx={{ fontSize: fontSize[12], fontWeight: 400 }}
+          >
+            {headCell.sortable ? (
+              <TableSortLabel
+                active={sortProperty === headCell.dataProperty}
+                direction={
+                  sortProperty === headCell.dataProperty
+                    ? sortDirection
+                    : SORT_DIRECTION.ASC
+                }
+                onClick={createSortHandler(headCell.dataProperty)}
+              >
+                {headCell.label}
+                {sortProperty === headCell.dataProperty ? (
+                  <Box component="span" sx={visuallyHidden}>
+                    {sortDirection === SORT_DIRECTION.DESC
+                      ? 'sorted descending'
+                      : 'sorted ascending'}
+                  </Box>
+                ) : null}
+              </TableSortLabel>
+            ) : (
+              headCell.label
+            )}
+          </TableCell>
+        ))}
+      </TableRow>
+    </TableHead>
+  )
+}
+
+const Table = (props) => {
+  const {
+    cellRenderer,
+    data,
+    headers,
+    sortDirection,
+    sortProperty,
+    handleRequestSort,
+  } = props
+
+  return (
+    <TableContainer component={Paper}>
+      <MuiTable size="small">
+        <EnhancedTableHead
+          sortDirection={sortDirection}
+          sortProperty={sortProperty}
+          onRequestSort={handleRequestSort}
+          headCells={headers}
+        />
+        <TableBody>
+          {data.map((rowData) => (
+            <TableRow>
+              {headers.map((header) => (
+                <TableCell key={header.dataProperty}>
+                  {cellRenderer(rowData, header.dataProperty)}
```

Comment:
this allows us to render custom elements for data properties. For example, we could render a participant's study as a link to that study or we can render star icons instead of the property value.